### PR TITLE
Initial commit of swagger design

### PR DESCRIPTION
**Pull request by floppyears**
CO-439

### CODE REVIEW

File: class-search-swagger.yaml [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+          type: string
+          description: "Whether or not the class meets on this day."
+        thursday:
+          type: string
+          description: "Whether or not the class meets on this day."
+        tuesday:
+          type: string
+          description: "Whether or not the class meets on this day."
+        wednesday:
+          type: string
```
Comment (magical) [inline class-search-swagger.yaml]: These should be bool. Also, why are all the days out of order?

File: term-swagger.yaml [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        description: "First day of classes in this term"
+      endDate:
+        type: string
+        description: "Last day of classes in this term"
+      financialAidYear:
+        type: string
+        description: "Aid year associated with this term"
+      housingStartDate:
+        type: string
+        description: "First day for meal plan and housing assignments"
```
Comment (magical) [inline term-swagger.yaml]: What date format?

File: academic-periods.yaml [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+# Describe your paths here
+paths:
+  # This is a path endpoint. Change it.
+  /academic-periods:
+    # This is a HTTP operation
+    get:
+      tags:
+       - academic-periods
+      # Describe this verb here. Note: you can use markdown
+      description: "Gets all `academic-level` objects. A combination of terms, years and subterms"
```
Comment (magical) [inline academic-periods.yaml]: s/academic-level/academic-period/

File: class-search-swagger.yaml [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+        "404":
+          description: "Not Found"
+          schema:
+            $ref: "#/definitions/Error"
+        "500":
+          description: "Internal Server Error"
+          schema:
+            $ref: "#/definitions/Errors"
+      security:
+        - locations_api_key: []
```
Comment (magical) [inline class-search-swagger.yaml]: Change to courses_api_key or delete?

File: class-search-swagger.yaml [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+
+    
+
+securityDefinitions:
+
+  locations_api_key:
```
Comment (magical) [inline class-search-swagger.yaml]: Change to courses_api_key or delete?

File: class-search-swagger.yaml [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+          schema:
+            $ref: "#/definitions/ResultObjects"
+        "404":
+          description: "Not Found"
+          schema:
+            $ref: "#/definitions/Error"
+        "500":
+          description: "Internal Server Error"
+          schema:
+            $ref: "#/definitions/Errors"
```
Comment (magical) [inline class-search-swagger.yaml]: Do you really want internal server error to return a list of errors?

File: class-search-swagger.yaml (hunk 1 above)
Comment (floppyears) [inline class-search-swagger.yaml]: Haha, because I was being a bit lazy yesterday at 6pm, I'll fix it :)

File: class-search-swagger.yaml (hunk 4 above)
Comment (floppyears) [inline class-search-swagger.yaml]: Good point. I used to specify to import the swagger and security information into the API gateway, but later on found out that the gateway didn't support this part of the swagger spec :(

File: class-search-swagger.yaml (hunk 6 above)
Comment (floppyears) [inline class-search-swagger.yaml]: Good catch!

File: class-search-swagger.yaml [hunk 7]
```diff
@@ -124,48 +113,44 @@ definitions:
...
-      seatsAvailable: 
-        type: string
-        description: "Number of seats available in the class."
       section: 
         type: string
         description: "Section class"
       status: 
         type: string
-        description: "Whether or not the course is open, closed"
+        description: "Whether or not the course is open or closed"
```
Comment (magical) [inline class-search-swagger.yaml]: s/Whether or not/Whether/

File: class-search-swagger.yaml [hunk 8]
```diff
@@ -124,48 +113,44 @@ definitions:
...
       subject:
         type: string
-        description: "Letter indification to specify the subject"
+        description: "Letter indification to specify the subject. This is the same value as the subject parameter"
```
Comment (magical) [inline class-search-swagger.yaml]: s/indification/identifier/ ?

File: class-search-swagger.yaml [hunk 9]
```diff
@@ -0,0 +1,250 @@
...
+        type: string
+        description: "Number of course. Sample value: `111` for MTH 111"
+      crn:
+        type: string
+        description: "Course Reference Number used to uniquely identify a class available for registration. This is the CRN value that students use to register for a class section."
+      sectionTitle:
+        type: string
+        description: "Title of the course section"
+      creditHourHigh:
+        type: string
```
Comment (magical) [inline class-search-swagger.yaml]: type: integer?

Same for creditHourLow, creditHours, enrollment, maximumEnrollment, waitCapacity, waitCount.

**Pull request merged**